libraries/standard/http/include/http_client.h

```diff
@@ -36,41 +37,77 @@
 #define HTTP_METHOD_HEAD    "HEAD"                 /**< HTTP Method HEAD. */
 
 /**
- * Flags for #HTTPRequestInfo_t.flags.
- * These flags control what headers are written or not to the
- * #HttpRequestHeaders_t.pBuffer.
+ * @section http_send_flags
+ * @brief Flags for #HTTPClient_Send parameter flags.
+ * These flags control some behavior of sending the request or receiving the
+ * response.
+ *
+ * - #HTTP_SEND_DISABLE_CONTENT_LENGTH_FLAG <br>
+ *   @copybrief HTTP_SEND_DISABLE_CONTENT_LENGTH_FLAG
+ *
+ * Flags should be bitwise-ORed with each other to change the behavior of
+ * #HTTPClient_Send.
  */
 
 /**
- * @brief Set this flag to indicate the request is for a persistent connection.
+ * @brief Set this flag to disable automatically sending the Content-Length
+ * header to the server.
  *
- * Setting this will cause a "Connection: Keep-Alive" to be written to the
- * request.
+ * This flag is valid only for #HTTPClient_Send.
  */
-#define HTTP_REQUEST_KEEP_ALIVE_FLAG                0x1U
+#define HTTP_SEND_DISABLE_CONTENT_LENGTH_FLAG    0x1U
 
 /**
- * @brief Set this flag to disable automatically writing the Content-Length
- * header.
+ * @section http_request_flags
+ * @brief Flags for #HTTPRequestInfo_t.flags.
+ * These flags control what headers are written or not to the
+ * #HttpRequestHeaders_t.pBuffer by #HTTPClient_InitializeRequestHeaders.
+ *
+ * - #HTTP_REQUEST_KEEP_ALIVE_FLAG <br>
+ *   @copybrief HTTP_REQUEST_KEEP_ALIVE_FLAG
+ *
+ * Flags should be bitwise-ORed with each other to change the behavior of
+ * #HTTPClient_InitializeRequestHeaders.
  */
-#define HTTP_REQUEST_DISABLE_CONTENT_LENGTH_FLAG    0x2U
 
 /**
- * Flags for #HTTPResponse_t.flags.
- * These flags are populated in #HTTPResponse_t.flags by the #HTTPClient_Send()
+ * @brief Set this flag to indicate the request is for a persistent connection.
+ *
+ * Setting this will cause a "Connection: Keep-Alive" to be written to the
+ * request headers.
+ * If this flag is not set, then "Connection: Close" will be written to the
+ * request headers.
+ *
+ * This flag is valid only for #HTTPRequestInfo.flags.
+ */
+#define HTTP_REQUEST_KEEP_ALIVE_FLAG    0x1U
```

Comment:
Since this is the only flag and only being used in `InitializeRequestHeaders`, maybe we can just have this as a bool as well.